### ISSUE TITLE
[BUG] Permission request on widget destroy

### ISSUE THREAD

**Issue by rasuru** [juliuscanute/qr_code_scanner#305]
**Describe the bug**
When navigating to a different screen, or when the user taps "Deny" on permission set and I want to show an alternative view that says no permissions where given (which means QrView is destroyed) the permission dialog pops up again. This is inconvenient because the user already denied the permission.

**Flutter information**
[✓] Flutter (Channel stable, 2.0.3, on Linux, locale en_US.UTF-8)
    • Flutter version 2.0.3 at /opt/flutter
    • Framework revision 4d7946a68d (4 weeks ago), 2021-03-18 17:24:33 -0700
    • Engine revision 3459eb2436
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/rasuru/.config/Android/Sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio
    • Android Studio at /opt/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Connected device (1 available)
    • SM J330F (mobile) • 4200a6aee0baa49d • android-arm • Android 9 (API 28)

! Doctor found issues in 1 category.

**Device (please complete the following information):**
 - Device: Samsung Galaxy J3
 - OS: Android 9

**Additional context**
I can reproduce the bug both when the StatefulWidget wrapper calls the dispose on QrViewController and when it does not.

**Comment by rasuru** [comment]
I fixed this using permission_handler and never rendering QRView until the user gives permission

**Comment by geisterfurz007** [comment]
This is reproducible starting in 0.3.3. The problem does not occur on 0.3.2.

**Comment by matiazar** [comment]
> I fixed this using permission_handler and never rendering QRView until the user gives permission

and how do you ask for permissions again to user if you dont render the QRView ?

thanks in advance

**Comment by rasuru** [comment]
@matiazar

I have a function that uses permission_handler:

```
Future requestCameraPermissions() async {
  final cameraStatus = await Permission.camera.status;

  if (cameraStatus.isGranted) {
    _hasPermissions$.update(true);
    return;
  }

  final newStatus = await Permission.camera.request();

  _hasPermissions$.update(newStatus.isGranted);
}
```

This function returns a future. I wrap the whole QRView in FutureBuilder, where the `future` argument is result of invoking this function.

`_hasPermissions$` is a state, inside the FutureBuilder I also have StreamBuilder that listens to changes to this state. Then it either shows a screen with QRView or a screen indicating no permissions were given and a button to prompt the user again.

**Comment by juliansteenbakker** [comment]
Merged #351 which fixes this problem
